app/controllers/tax_returns_controller.rb

```diff
@@ -0,0 +1,54 @@
+class TaxReturnsController < ApplicationController
+  # Because these aren't CRUD verb methods, they need to be explicitly listed on the before actions
+  before_action :block_access_on_production, only: [:authorize_signature, :sign, :success]
+  before_action :load_tax_return, except: [:success]
+  before_action :check_for_forms, only: [:sign, :authorize_signature]
+
+  def authorize_signature
+    tax_return = TaxReturn.find(params[:tax_return_id])
+    @form = Portal::SignForm8879.new(tax_return)
+  end
+
+  def sign
+    @form = Portal::SignForm8879.new(@tax_return, permitted_params)
+    if @form.valid? && @form.sign
+      redirect_to tax_return_success_path(params[:tax_return_id])
+    else
+      flash.now[:alert] = I18n.t("controllers.tax_returns_controller.errors.#{@form.errors.keys.first}")
```

Comment:
should this be the first error message or all? i think there's a `.full_messages` or some such but idk whether it will string them together in a nice way

Reply:
I think it's fine to just show one flash error for both checkboxes at the top.

The way this is set up, it will put error messages on both check box fields when it validates, but it will add a more detailed error as a flash alert.